lldb/source/Plugins/RegisterTypeBuilder/RegisterTypeBuilderClang.cpp

```diff
@@ -116,7 +116,8 @@ CompilerType RegisterTypeBuilderClang::GetRegisterType(
     type_system->SetIsPacked(fields_type);
 
     // This should be true if RegisterFlags padded correctly.
-    assert(*fields_type.GetByteSize(nullptr) == flags.GetSize());
+    assert(llvm::expectedToOptional(fields_type.GetByteSize(nullptr))
+               .value_or(0) == flags.GetSize());
```

Comment:
Just so it's consistent with the previous version:
```suggestion
    assert(llvm::expectedToOptional(fields_type.GetByteSize(nullptr)).value() == flags.GetSize());
```

Reply:
See my other comment, I think this is actually the original intent of the code.